backend/tracim_backend/lib/webdav/resources.py

```diff
@@ -68,143 +70,224 @@ def wrapper(self: "_DAVResource", *arg, **kwarg) -> typing.Callable:
+            )
+        return members_names
 
-        We don't allow to create new workspaces through
-        webdav client.
+    def getMember(self, label: str) -> typing.Optional[_DAVResource]:
         """
-        raise DAVError(HTTP_FORBIDDEN, contextinfo="Not allowed to create item in the root dir")
+        Access to a specific members
+        """
+        return self.provider.getResourceInst(
+            "%s/%s" % (self.path, webdav_convert_file_name_to_display(label)), self.environ
+        )
 
-    @webdav_check_right(is_user)
     def getMemberList(self):
         """
         This method is called by wsgidav when requesting with a depth > 0, it will return a list of _DAVResource
         of all its direct children
         """
-
-        members_names = []  # type: List[str]
         members = []
-        for workspace in self.workspace_api.get_all():
-            # fix path
-            workspace_label = workspace.label
-            if workspace_label in members_names:
-                # INFO - G.M - 2020-09-24
-                # We decide to show only first same name workspace (in order of get_all() which is workspace
-                # id order). So, same name workspace are not supported from webdav but doesn't cause
-                # much trouble: only one workspace is accessible without any issue.
-                continue
-            path = add_trailing_slash(self.path)
-            # return item
-            workspace_path = "{}{}".format(path, workspace_label)
+        for workspace in self._get_members():
             members.append(
-                WorkspaceResource(
-                    path=workspace_path,
-                    environ=self.environ,
-                    workspace=workspace,
-                    tracim_context=self.tracim_context,
-                    label=workspace_label,
+                self._generate_child_workspace_resource(
+                    parent_path=self.path, child_workspace=workspace
                 )
             )
-            members_names.append(workspace_label)
         return members
 
 
-class AbstractContentContainer(DAVCollection, ABC):
+class ContentOnlyContainer(WebdavContainer):
+    """
+    Container that can get children content
+    """
+
     def __init__(
         self,
         path: str,
         environ: dict,
         label: str,
+        content: Content,
+        provider: "TracimDavProvider",
         workspace: Workspace,
         tracim_context: "WebdavTracimContext",
     ) -> None:
-        super(AbstractContentContainer, self).__init__(path, environ)
+        """
+        Some rules:
+        - if content given is None, return workspace root contents
+        - if content given is correct, return subcontent of this content
+         add user known workspaces without any user known parent to the list.
+         - in case of content collision, only the first named content (sorted by content_id
+         from lower to higher) will be returned.
```

Comment:
```suggestion
        - if content given is None, return workspace root contents
        - if the given content is correct, return the subcontent of this content
         and user-known workspaces without any user-known parent to the list.
         - in case of content collision, only the first named content (sorted by content_id
         from lower to higher) will be returned.
```